hack/e2e-test.sh

```diff
@@ -53,64 +53,101 @@ make deploy DEPLOY_IMAGE="${HIVE_IMAGE}"
+	if mkdir -p "${LOGS_DIR}"
+	then
+		if INSTALL_JOB_NAME=$(oc get job -l "hive.openshift.io/cluster-deployment-name=${CLUSTER_NAME},hive.openshift.io/install=true" -o name) && [ "${INSTALL_JOB_NAME}" ]
+		then
+			oc logs -c hive job/${INSTALL_JOB_NAME} &> "${LOGS_DIR}/hive_install_job.log" || true
+			oc get job ${INSTALL_JOB_NAME} -o yaml &> "${LOGS_DIR}/hive_install_job.yaml" || true
+		fi
+		oc get clusterdeployment -A -o yaml &> "${LOGS_DIR}/hive_clusterdeployment.yaml" || true
+		oc get clusterimageset -o yaml &> "${LOGS_DIR}/hive_clusterimagesets.yaml" || true
+		echo "************* INSTALL JOB LOG *************"
+		if oc get clusterprovision -l "hive.openshift.io/cluster-deployment-name=${CLUSTER_NAME}" -o jsonpath='{.items[0].spec.installLog}' &> "${LOGS_DIR}/hive_install_console.log"; then
+			cat "${LOGS_DIR}/hive_install_console.log"
+		else
+			cat "${LOGS_DIR}/hive_install_job.log"
+		fi
 	fi
 
 	echo ""
 	echo ""
 	echo "Deleting ClusterDeployment ${CLUSTER_NAME}"
-	oc delete --wait=false clusterdeployment ${CLUSTER_NAME}
-	errorOnUninstall=0
-	if ! go run "${SRC_ROOT}/contrib/cmd/waitforjob/main.go" --log-level=debug "${CLUSTER_NAME}" "uninstall"; then
-		errorOnUninstall=1
-	fi
+	oc delete --wait=false clusterdeployment ${CLUSTER_NAME} || :
 
-	if [[ $errorOnUninstall == 1 ]]; then
-		if oc logs job/${CLUSTER_NAME}-uninstall &> "${ARTIFACT_DIR}/hive_uninstall_job.log"; then
+	if ! go run "${SRC_ROOT}/contrib/cmd/waitforjob/main.go" --log-level=debug "${CLUSTER_NAME}" "uninstall"
+	then
+		echo "Waiting for uninstall job failed"
+		if mkdir -p "${LOGS_DIR}" && oc logs job/${CLUSTER_NAME}-uninstall &> "${LOGS_DIR}/hive_uninstall_job.log"
+		then
 			echo "************* UNINSTALL JOB LOG *************"
 			cat "${ARTIFACT_DIR}/hive_uninstall_job.log"
 			echo ""
 			echo ""
-		else
-			echo "Waiting for uninstall job failed"
 		fi
 		exit 1
 	fi
 }
-trap 'teardown' EXIT
 
 echo "Running post-deploy tests"
 make test-e2e-postdeploy
 
-echo "Creating cluster deployment"
-echo "Release image: $RELEASE_IMAGE"
-SRC_ROOT=$(git rev-parse --show-toplevel)
-
-go run "${SRC_ROOT}/contrib/cmd/hiveutil/main.go" create-cluster "${CLUSTER_NAME}" \
-	--ssh-public-key-file="${SSH_PUBLIC_KEY_FILE}" \
-	--pull-secret-file="${PULL_SECRET_FILE}" \
-	--base-domain="${BASE_DOMAIN}" \
-	--release-image="${RELEASE_IMAGE}" \
-	--install-once=true \
-	--uninstall-once=true
+export SRC_ROOT=$(git rev-parse --show-toplevel)
+
+for PLATFORM in "${PLATFORMS[@]}"
+do
+	CLUSTER_NAME=${BASE_CLUSTER_NAME}-${PLATFORM}
+	case "$PLATFORM" in
+	"aws")
+		EXTRA_CREATE_CLUSTER_ARGS=" --base-domain=${AWS_BASE_DOMAIN} --creds-file=${CLOUD_CREDS_DIR}/.awscred"
+		;;
+	"azure")
+		EXTRA_CREATE_CLUSTER_ARGS=" --base-domain=${AZURE_BASE_DOMAIN} --creds-file=${CLOUD_CREDS_DIR}/osServicePrincipal.json"
+		;;
+	"gcp")
+		EXTRA_CREATE_CLUSTER_ARGS=" --base-domain=${GCP_BASE_DOMAIN} --creds-file=${CLOUD_CREDS_DIR}/gce.json --gcp-project-id=openshift-gce-devel-ci"
```

Comment:
What's getting used for GCE and Azure credentials when we run this? Something available to all e2e jobs?

Reply:
I was hoping that the credentials would be there from these variables.
https://github.com/openshift/release/blob/dfd2beadbd400ce3ac680b2e3a430f58bcc5e63d/ci-operator/templates/openshift/installer/cluster-launch-installer-src.yaml#L157-L162
But it appears that the credentials are only created for the platform that the e2e tests are being run on. I need to dig into this more to see if and how we can get all of the credentials.

Reply:
In light of the credentials problem, I think I will abandon the multiple-platforms-in-one-e2e-test-suite approach in favor of running multiple e2e test suites.

Reply:
It is unfortunate that this will necessitate creating twice as many clusters, though.